azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_servicefabric.go

```diff
@@ -0,0 +1,276 @@
+				"server_certificate_common_name": servicefabricServerCertificateCommonNameSchema(resourceBlockServiceFabricCertificate),
+				"client_certificate": {
+					Type:             schema.TypeString,
+					Required:         true,
+					Description:      "Base64 encoding of the cluster's client certificate file.",
+					Sensitive:        true,
+					ValidateFunc:     validation.StringIsNotEmpty,
+					DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+				},
+				"client_certificate_password": {
+					Type:             schema.TypeString,
+					Optional:         true,
+					Description:      "Password for the certificate.",
+					Sensitive:        true,
+					ValidateFunc:     validation.StringIsNotEmpty,
+					DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+				},
+				secretHashKeyClientCertificate:         secretHashSchemaClientCertificate,
+				secretHashKeyClientCertificatePassword: secretHashSchemaClientCertificatePassword,
+			},
+		},
+		ConflictsWith: []string{resourceBlockServiceFabricAzureActiveDirectory, resourceBlockServiceFabricNone},
+	}
+
+	secretHashKeyPassword, secretHashSchemaPassword := tfhelper.GenerateSecreteMemoSchema("password")
+	r.Schema[resourceBlockServiceFabricAzureActiveDirectory] = &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"server_certificate_lookup":      servicefabricServerCertificateLookupSchema(),
+				"server_certificate_thumbprint":  servicefabricServerCertificateThumbprintSchema(resourceBlockServiceFabricAzureActiveDirectory),
+				"server_certificate_common_name": servicefabricServerCertificateCommonNameSchema(resourceBlockServiceFabricAzureActiveDirectory),
+				"username": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+					Description:  "Specify an Azure Active Directory account.",
+				},
+				"password": {
+					Type:             schema.TypeString,
+					Required:         true,
+					Description:      "Password for the Azure Active Directory account.",
+					Sensitive:        true,
+					ValidateFunc:     validation.StringIsNotEmpty,
+					DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+				},
+				secretHashKeyPassword: secretHashSchemaPassword,
+			},
+		},
+		ConflictsWith: []string{resourceBlockServiceFabricCertificate, resourceBlockServiceFabricNone},
+	}
+
+	r.Schema[resourceBlockServiceFabricNone] = &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"unsecured": {
+					Type:        schema.TypeBool,
+					Optional:    true,
+					Default:     false,
+					Description: "Skip using windows security for authentication.",
+				},
+				"cluster_spn": {
+					Type:         schema.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
`cluster_spn` accept empty string, Use validation like  `validation.StringLenBetween(0, 1024),` check the value length is the better choice.